src/common/updating/http_client.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <future>
+#include <winrt/Windows.Web.Http.h>
+
+namespace http
+{
+    class HttpClient
+    {
+    public:
+        HttpClient();
+        std::future<std::wstring> request(const winrt::Windows::Foundation::Uri& url);
+        std::future<void> download(const winrt::Windows::Foundation::Uri& url, const std::wstring& dstFle);
+        std::future<void> download(const winrt::Windows::Foundation::Uri& url, const std::wstring& dstFle, const std::function<void(float)>& progressUpdateHandle);

```

Comment:
`progressUpdateHandle` -> `progressUpdateCallback`. we could also pass it by value and perform `std::move` inside.